ledger/babbage/rules.go

```diff
@@ -184,10 +184,21 @@ func UtxoValidateCollateralEqBalance(
 		}
 		collBalance += utxo.Output.Amount()
 	}
-	// Subtract collateral return amount
+
+	// Skip validation if no valid collateral UTxOs were found
+	// This avoids subtracting from zero and prevents uint underflow
+	if collBalance == 0 {
+		return nil
+	}
+
+	// Subtract collateral return amount with underflow protection
 	collReturn := tx.CollateralReturn()
 	if collReturn != nil {
-		collBalance -= collReturn.Amount()
+		returnAmt := collReturn.Amount()
+		if collBalance < returnAmt {
+			return errors.New("collateral return amount exceeds collateral input balance")
```

Comment:
We only want to return the specific errors defined in the ledger for each era, and generally only at the end of the function